diff --git a/zephyr-build/src/lib.rs b/zephyr-build/src/lib.rs
index 8199cb3..28278fb 100644
--- a/zephyr-build/src/lib.rs
+++ b/zephyr-build/src/lib.rs
@@ -66,9 +66,9 @@ pub fn build_kconfig_mod() {
         if let Some(caps) = config_hex.captures(&line) {
             writeln!(&mut f, "#[allow(dead_code)]").unwrap();
-            writeln!(&mut f, "pub const {}: usize = {};", &caps[1], &caps[2]).unwrap();
+            writeln!(&mut f, "pub const {}: u64 = {};", &caps[1], &caps[2]).unwrap();
         } else if let Some(caps) = config_int.captures(&line) {
             writeln!(&mut f, "#[allow(dead_code)]").unwrap();
-            writeln!(&mut f, "pub const {}: isize = {};", &caps[1], &caps[2]).unwrap();
+            writeln!(&mut f, "pub const {}: i64 = {};", &caps[1], &caps[2]).unwrap();
         } else if let Some(caps) = config_str.captures(&line) {
             writeln!(&mut f, "#[allow(dead_code)]").unwrap();
             writeln!(&mut f, "pub const {}: &str = {};", &caps[1], &caps[2]).unwrap();
